app/presenters/detailed_guide_presenter.rb

```diff
@@ -0,0 +1,81 @@
+class DetailedGuidePresenter < ContentItemPresenter
+  include ExtractsHeadings
+  include ActionView::Helpers::UrlHelper
+
+  def body
+    content_item["details"]["body"]
+  end
+
+  def contents
+    extract_headings_with_ids(body).map do |heading|
+      link_to(heading[:text], "##{heading[:id]}")
+    end
+  end
+
+  def context
+    parent["title"]
+  end
+
+  def breadcrumbs
+    [
+      { title: "Home", url: "/" },
+      { title: context, url: parent["base_path"] }
+    ]
+  end
+
+  def from
+    links("lead_organisations") + links("supporting_organisations") + links("worldwide_organisations")
```

Comment:
Some of these bits were copied from case studies (published, from, part of, history), could the shared parts be refactored into helpers/modules, maybe like https://github.com/alphagov/government-frontend/blob/master/app/presenters/extracts_headings.rb